pkg/apis/config/doc.go

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2020 The Multicluster-Scheduler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// +k8s:deepcopy-gen=package
+// +groupName=kubescheduler.config.k8s.io
```

Comment:
Why is this not `config.admiralty.io`?

Taking a step back, why don't you simply add the config types to `pkg/apis/multicluster`?

If you do need a separate group, declaring a v1 seems premature.

Reply:
So it's my understanding that for out of tree scheduler plugins, it has to be registered to the same GVK as the scheduler configs since the loading of those hardcodes and expects to parse as the plugin name + "Args" when converting to Go types.

https://github.com/kubernetes/kubernetes/blob/4fa7ce6f3c0038ecec7c2ec3d6edb42315db3015/staging/src/k8s.io/kube-scheduler/config/v1/types.go#L287-L289

https://github.com/kubernetes/kubernetes/blob/4fa7ce6f3c0038ecec7c2ec3d6edb42315db3015/pkg/scheduler/apis/config/v1/defaults.go#L89-L91


Since 1.29, [all alphas/betas are deprecated](https://github.com/kubernetes/kubernetes/blob/43a444e7fb503786ed2cd2ad1de788197c6b895b/cmd/kube-scheduler/app/options/configfile.go#L68-L72) so only the v1 is supported. `kubescheduler.config.k8s.io/v1` is supported [since 1.25](https://github.com/kubernetes/kubernetes/blob/88e994f6bf8fc88114c5b733e09afea339bea66d/cmd/kube-scheduler/app/options/configfile.go#L77-L78).